atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/KnownConcludedTransactionsStore.java

```diff
@@ -117,6 +121,27 @@ public void supplement(Set<Range<Long>> timestampRangesToAdd) {
         throw new SafeIllegalStateException("Unable to supplement set of concluded timestamps.");
     }
 
+    /**
+     * Modifies a range that contains the minimum timestamp to have it's lower-bound set (exclusive) to the minimum timestamp.
+     *
+     * This is needed as we do not want to not conclude a range that encompasses the minimum timestamp.
+     * We also do not want to prevent our ability to conclude the subrange which is above the minimum timestamp.
+     */
+    @VisibleForTesting
+    Range<Long> maybeModifyRangeLowerBoundToMinimumTimestamp(Optional<Long> maybeMinimumTimestamp, Range<Long> range) {
+        return maybeMinimumTimestamp
+                .filter(range)
+                .map(minimumTimestamp -> Range.openClosed(minimumTimestamp, range.upperEndpoint()))
+                .orElse(range);
+    }
+
+    private boolean maybeFilterRangeIfUpperBoundLessThanMinimumTimestamp(
+            Optional<Long> maybeMinimumTimestamp, Range<Long> range) {
```

Comment:
yep instead of having two methods we can just do one that does range.intersect([minTs + 1, inf]) and leverage the built in intersect util.